### PR TITLE
Unstyled messages frontend (relies on anika/messages/backend)

### LINKED ISSUE #75: 𝗣𝗮𝗴𝗲: Create Messaging Page
## Feature Description

**Basic Messaging: Users should be able to message other users by replying on a listing and have an inbox view of previous chats.**
This page is for facilitating communication between users and sellers, each coversation in messages is associated with a listing, and this will need to be reflected in the Mockups. Coversations are only between two people and will be using Websockets for live messaging.

This should also wait until later in the sprint, as we will likely need some help from @UVicMartletplace/backend for implementing this. However, it can likely be mocked relatively easily. 

This requires components of:
- #42 
- #45 
- TBD

relates: #45, #42, #55
blocked by: #42, #45

### PR DESCRIPTION
**Pull request by abookout**
# Description

- Includes some very minor global style changes (App.css wasn't actually being used anyway so set them to a sane default)
- ~~user_id is hardcoded to "5" until auth integration done~~
- ~~tests are minimal (will write more tomorrow)~~

Closes #75

## Checklist
- [ x ] The code includes tests if relevant
- [ x ] I have *actually* self-reviewed my changes and done QA

### CODE REVIEW

**Comment by CaelumD25** [comment]
I'd love to get this merged before lab today, but I see that it's not passing quite a few checks, including build. Is it possible to get these fixed before demoing in the lab today?

File: apps/frontend/src/pages/messages/ConversationsSidebar.tsx [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+              <Box
+                component="img"
+                src={thread.other_participant.profile_pic_url}
+                sx={s.messagesConvImg}
+              />
+            </button>
+          );
+        })}
+      </Stack>
+    </Box>
```
Comment (CaelumD25) [inline apps/frontend/src/pages/messages/ConversationsSidebar.tsx:58]: Can we get this swapped to this?
```suggestion
<Box id="conversations_sidebar" sx={s.messagesConvSidebar}>
      <Stack direction="row">
        <Typography variant="h5">Conversations</Typography>
      </Stack>
      <Stack direction="column">
        {threads.map((thread) => {
          const isThreadSelected =
            selectedThread !== null &&
            selectedThread.listing_id === thread.listing_id &&
            selectedThread.other_participant.user_id ===
            thread.other_participant.user_id;
          return (
            <button
              key={
                "" +
                thread.listing_id +
                thread.other_participant.user_id +
                user_id
              }
              // TODO: add a 'selected' class, this is just to show how to apply the style conditionally
              style={{
                ...s.messagesConvBox,
                ...(isThreadSelected ? { backgroundColor: "white" } : {}),
              }}
              onClick={() => selectThread(thread)}
            >
              <div style={s.messagesConvPreviewText}>
                <Typography variant="body1">
                  {thread.last_message.content}
                </Typography>
              </div>
              <Box
                component="img"
                src={thread.other_participant.profile_pic_url}
                sx={s.messagesConvImg}
              />
            </button>
          );
        })}
      </Stack>
    </Box>
```

File: apps/frontend/src/pages/messages/MessageSendBox.tsx [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+            type="submit"
+            size="small"
+            onClick={onClickSend}
+            sx={s.button}
+          >
+            Send
+          </Button>
+        </Stack>
+      </form>
+    </Box>
```
Comment (CaelumD25) [inline apps/frontend/src/pages/messages/MessageSendBox.tsx:46]: Can we get this swapped with this?

```suggestion
<Box sx={s.messagesSendBox}>
      <form
        onSubmit={(e) => {
          e.preventDefault();
        }}
      >
        <Box
          sx={{
            display: "flex",
            flexDirection: "row",
            justifyContent: "space-between",
          }}
        >
          <Input
            onChange={onType}
            placeholder="Type Here"
            value={text}
            sx={{
              flex: "4",
              borderColor: "black",
              border: "10px",
              borderRadius: "10px",
            }}
          />
          <Button
            type="submit"
            onClick={onClickSend}
            sx={{ ...s.button, flex: "1", color:"white", marginLeft: "10px" }}
          >
            Send
          </Button>
        </Box>
      </form>
    </Box>
```

File: apps/frontend/src/pages/messages/Message.tsx [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+  return (
+    <Box
+      sx={
+        message.sender_id == user.user?.userID
+          ? s.messageFromUser
+          : s.messageFromOther
+      }
+    >
+      {message.message_body}
+    </Box>
```
Comment (CaelumD25) [inline apps/frontend/src/pages/messages/Message.tsx:21]: Can we get this swapped to this?

```suggestion
<Box
      sx={
        message.sender_id == user.user?.userID
          ? s.messageFromUser
          : s.messageFromOther
      }
    >
      <Typography variant="body2">{message.message_body}</Typography>
    </Box>
```

File: apps/frontend/src/pages/messages/Messages.tsx [hunk 4]
```diff
@@ -0,0 +1,263 @@
...
+                    ))}
+                  </InfiniteScroll>
+                </Box>
+                <MessageSendBox onMessageSend={onMessageSend} />
+              </Box>
+            )}
+          </Stack>
+        </Box>
+      )}
+    </Box>
```
Comment (CaelumD25) [inline apps/frontend/src/pages/messages/Messages.tsx:259]: Can we get this swapped?

```suggestion
<Box>
      <Box ref={headerRef}>
        <SearchBar />
      </Box>
      <Container>
        {loading ? (
          <CircularProgress />
        ) : threads.length === 0 ? (
          <Box
            sx={{
              display: "flex",
              justifyContent: "center",
              alignItems: "center",
            }}
          >
            <p>
              No threads yet! Find a listing you like and begin a conversation
              from there.
            </p>
          </Box>
        ) : (
          <Box
            sx={{
              position: "relative",
              height: `calc(100vh - ${headerHeight}px)`,
            }}
          >
            {isMobileSize && currentThread !== null && (
              <Button sx={s.messagesHideThreadButton} onClick={hideThread}>
                Back
              </Button>
            )}
            <Stack direction="row">
              {(!isMobileSize || currentThread === null) && (
                <ConversationsSidebar
                  threads={threads}
                  selectThread={selectThread}
                  selectedThread={currentThread}
                />
              )}
              {shouldShowMessages && (
                <Box
                  sx={{
                    // Contains both the messages and the send box
                    width: "100%",
                    height: `calc(100vh - ${vars.messagesSendBoxHeight} - ${headerHeight}px)`,
                  }}
                >
                  <Box sx={s.messagesMessagesBox} id="scrollable">
                    <InfiniteScroll
                      load={() =>
                        fetchMoreMessages(currentThread, messagesState.length)
                      }
                      hasMore={hasMoreMessages}
                      loader={
                        loading && (
                          <CircularProgress
                            size={"2rem"}
                            sx={{ marginHorizontal: "auto" }}
                          />
                        )
                      }
                      scrollContainerId={"scrollable"}
                    >
                      {messagesState.map((item) => (
                        <Message message={item} key={item.message_id} />
                      ))}
                    </InfiniteScroll>
                  </Box>
                  <MessageSendBox onMessageSend={onMessageSend} />
                </Box>
              )}
            </Stack>
          </Box>
        )}
      </Container>
    </Box>
```

File: apps/frontend/src/styles/pageStyles.tsx [hunk 5]
```diff
@@ -179,4 +183,88 @@ export const useStyles = () => ({
     fontSize: "16px",
     height: "80px",
   },
+
+  // Messages page styles
+  messagesConvSidebar: {
+    overflowY: "scroll",
```
Comment (CaelumD25) [inline apps/frontend/src/styles/pageStyles.tsx:189]:
```suggestion
    overflowY: "auto",
```

File: apps/frontend/src/styles/pageStyles.tsx [hunk 6]
```diff
@@ -179,4 +183,88 @@ export const useStyles = () => ({
...
+    width: "20rem",
+    height: "100%",
+    borderRight: "2px solid",
+    borderColor: colors.martletplaceGrey,
```
Comment (CaelumD25) [inline apps/frontend/src/styles/pageStyles.tsx:193]:
```suggestion

```

File: apps/frontend/src/styles/pageStyles.tsx [hunk 7]
```diff
@@ -179,4 +183,88 @@ export const useStyles = () => ({
...
+    "@media (max-width: 740px)": {
+      width: "100%",
+    },
```
Comment (CaelumD25) [inline apps/frontend/src/styles/pageStyles.tsx:196]:
```suggestion
  maxHeight: "300px",
    "&::-webkit-scrollbar": {
       width: "4px",
     },
     "&::-webkit-scrollbar-thumb": {
       backgroundColor: colors.martletplaceGrey,
       borderRadius: "2px",
     },
     "&::-webkit-scrollbar-track": {
      backgroundColor: "white",
    },
```

File: apps/frontend/src/styles/pageStyles.tsx [hunk 8]
```diff
@@ -179,4 +183,88 @@ export const useStyles = () => ({
...
+  },
+  messagesMessagesBox: {
+    height: "100%",
+    overflowY: "scroll",
+
+    display: "flex",
```
Comment (CaelumD25) [inline apps/frontend/src/styles/pageStyles.tsx:202]:
```suggestion
    display: "flex",
    borderBottom: "1px",
    borderColor: colors.martletplaceGrey,
```

File: apps/frontend/src/styles/pageStyles.tsx [hunk 9]
```diff
@@ -179,4 +183,88 @@ export const useStyles = () => ({
...
+    flexDirection: "column-reverse" as const,
+    alignItems: "center",
+    gap: "1rem",
+
```
Comment (CaelumD25) [inline apps/frontend/src/styles/pageStyles.tsx:206]:
```suggestion
    "&::-webkit-scrollbar": {
      width: "4px",
    },
    "&::-webkit-scrollbar-thumb": {
      backgroundColor: colors.martletplaceGrey,
      borderRadius: "2px",
    },
    "&::-webkit-scrollbar-track": {
      backgroundColor: "white",
    },
```

File: apps/frontend/src/styles/pageStyles.tsx [hunk 10]
```diff
@@ -179,4 +183,88 @@ export const useStyles = () => ({
...
+    "& :first-of-type": {
```
Comment (CaelumD25) [inline apps/frontend/src/styles/pageStyles.tsx:207]:
```suggestion
    "& :first-child": {
```

File: apps/frontend/src/styles/pageStyles.tsx [hunk 11]
```diff
@@ -179,4 +183,88 @@ export const useStyles = () => ({
...
+      marginBottom: "0.25rem",
+    },
+  },
+  messagesSendBox: {
+    height: vars.messagesSendBoxHeight,
+    borderTop: "2px solid",
+    borderColor: colors.martletplaceGrey,
```
Comment (CaelumD25) [inline apps/frontend/src/styles/pageStyles.tsx:214]:
```suggestion

```

File: apps/frontend/src/styles/pageStyles.tsx [hunk 12]
```diff
@@ -179,4 +183,88 @@ export const useStyles = () => ({
...
+    color: colors.martletplaceGrey,
+  },
+  messagesSendButton: {},
+  messageFromUser: {
+    alignSelf: "flex-end",
+    backgroundColor: colors.martletplaceGrey,
+    color: colors.martletplaceWhite,
```
Comment (CaelumD25) [inline apps/frontend/src/styles/pageStyles.tsx:221]:
```suggestion
      backgroundColor: colors.martletplaceSkyBlue,
```

File: apps/frontend/src/styles/pageStyles.tsx [hunk 13]
```diff
@@ -179,4 +183,88 @@ export const useStyles = () => ({
...
+    padding: "0.5rem",
```
Comment (CaelumD25) [inline apps/frontend/src/styles/pageStyles.tsx:222]:
```suggestion
    padding: "10px",
```

File: apps/frontend/src/styles/pageStyles.tsx [hunk 14]
```diff
@@ -179,4 +183,88 @@ export const useStyles = () => ({
...
+    maxWidth: "80vw",
+    borderRadius: "10px",
+    textAlign: "right",
+    borderBottomRightRadius: "2px",
+    marginInline: "1rem",
+  },
+  messageFromOther: {
+    alignSelf: "flex-start",
+    backgroundColor: colors.martletplaceSkyBlue,
```
Comment (CaelumD25) [inline apps/frontend/src/styles/pageStyles.tsx:231]:
```suggestion
    backgroundColor: colors.martletplaceGrey,
```

File: apps/frontend/src/styles/pageStyles.tsx [hunk 15]
```diff
@@ -179,4 +183,88 @@ export const useStyles = () => ({
...
+    color: colors.martletplaceWhite,
+    padding: "0.5rem",
```
Comment (CaelumD25) [inline apps/frontend/src/styles/pageStyles.tsx:233]:
```suggestion
    padding: "10px",
```

File: apps/frontend/src/styles/pageStyles.tsx [hunk 16]
```diff
@@ -179,4 +183,88 @@ export const useStyles = () => ({
...
+  },
+  messagesConvBox: {
+    display: "flex",
+    flexDirection: "row" as const,
+    alignItems: "center",
+    justifyContent: "space-between",
+    gap: "1rem",
+    padding: "0.5rem",
+    borderBottom: "1px solid",
+    borderColor: colors.martletplaceGrey,
```
Comment (CaelumD25) [inline apps/frontend/src/styles/pageStyles.tsx:248]:
```suggestion
    padding: "10px",
    border: "none",
    backgroundColor: "#fcfcfc",
```

**Review by CaelumD25 — APPROVED** [review]
Seems to be working for me, I don't understand all of the code, but I can't see anything that I think would be breaking.

I've made suggestions for quick fixes to the styling so you should just be able to accept them on github with minimal work, I would've done this differently but my laptop isn't happy.

File: apps/frontend/src/styles/pageStyles.tsx (hunk 10 above)
Comment (abookout) [reply apps/frontend/src/styles/pageStyles.tsx:207]: this would give a console error

**Pull request merged**